controller/volume_controller.go

```diff
@@ -762,7 +762,7 @@ func (vc *VolumeController) cleanupReplicas(v *longhorn.Volume, es map[string]*l
 	// 	since the getHealthyReplicaCount function doesn't differentiate between replicas of different engines
 	// 	then during cleanupExtraHealthyReplicas the condition `healthyCount > v.Spec.NumberOfReplicas` will be true
 	//  which can lead to incorrect deletion of replicas.
-	if vc.isVolumeMigrating(v) || vc.isVolumeUpgrading(v) {
+	if vc.isVolumeMigrating(v) {
```

Comment:
NIT: Need to modify the above comment a little bit. Besides, will executing `cleanupExtraHealthyReplicas` for upgrading volumes lead to any issues?

Reply:
Add some comment to explain it.
I am afraid of `DataLocality` and `AutoBalanced` might lead some issues potentaily in `cleanupExtraHealthyReplicas`

Reply:
Yes. Please test these features with the PR. Besides, you may need to check the live migration feature as well.

Reply:
`cleanupExtraHealthyReplicas` is the last statement in `cleanupReplicas`, so theriotically, it should be fine, because no healthy replicas will be impacted during the volume upgrade. Need to follow up with QA later, cc @longhorn/qa